### ISSUE TITLE
[devel_transport] I should be able to use this module without utest behind scenes

### ISSUE THREAD

**Issue by PrzemekWirkus** [ARMmbed/unity#2]
Niklas,
I've been porting `minar-tests` and I wanted to use only `unity` asserts in test cases.
My module dependencies are:

``` json
{
  "dependencies": {
    "utest": "^1.5.0"
  }
}
```

When using this unity yotta module as test dependency I do not expect it to first of all drag `utest` to the game, second I do not expect it to print:
- `utest` stuff: `>>> ...` or 
- `greentea` stuff: `__testcase_summary`.

Comments:
- Using asserts with this module ALWAYS causes extra `utest` prints! This is not good because we are capturing in greentea extra valid prints.
- `utest` prints should not be turn on when we are not initializing harness!
## Greentea support for non utest test suites

Greenta does support non-utest test suites!! It is a feature, idiom looks like this:

``` c++

#include "greentea-client/test_env.h"  // GREENTEA_SETUP, GREENTEA_TESTSUITE_RESULT
#include "unity/unity.h"  // !Optional! TEST_ASSERT_*

// Inside test case
int main() {
    bool result = false;
    GREENTEA_SETUP(10, "default");

    // do your testing
    // ... use unity asserts:

    TEST_ASSERT_TRUE_MESSAGE(false, "boom!");

    // more test stuff...

    GREENTEA_TESTSUITE_RESULT(result);
}
```
## Test case example (prints utest stuff

This test case doesn't initialize `Harness` so I do not expect it to print utest messages.

``` c++
#include "mbed-drivers/mbed.h"
#include "minar/minar.h"
#include "greentea-client/test_env.h"
#include "unity/unity.h"

static void regular_callback ()
{
    minar::tick_t now = minar::platform::getTime();

    if ( now >= CALLBACK_TIME and now <= CALLBACK_TIME + TOLERANCE ) {
        GREENTEA_TESTSUITE_RESULT(true);
    } else {
        GREENTEA_TESTSUITE_RESULT(false);
    }
}

static void schedule_callback ()
{
    minar::tick_t now = minar::platform::getTime();
    minar::tick_t delay = CALLBACK_TIME - now;

    minar::Scheduler::postCallback(mbed::util::FunctionPointer0<void>(regular_callback).bind())
        .delay((minar::tick_t) delay)
        .tolerance(0);

    TEST_ASSERT_TRUE(now >= SCHEDULE_TIME);
}

void app_start(int, char*[])
{
    // Schedule a non-periodic callback when time is 0xffflarge to be called at 0x0small just after timer overflow
    GREENTEA_SETUP(120, "default");

    minar::tick_t now  = minar::platform::getTime();
    minar::tick_t delay = 0;

    if (now < SCHEDULE_TIME) {
        delay = SCHEDULE_TIME - now;
    }

    minar::Scheduler::postCallback(mbed::util::FunctionPointer0<void>(schedule_callback).bind())
        .delay((minar::tick_t) delay)
        .tolerance(0);
}
```
## Some log to check:

```
[1456529865.28][HTST][INF] copy image onto target...                                                              
        1 file(s) copied.                                                                                         
[1456529872.95][HTST][INF] starting host test process...                                                          
[1456529873.52][CONN][INF] starting connection process...                                                         
[1456529873.52][CONN][INF] initializing serial port listener...                                                   
[1456529873.52][SERI][INF] serial(port=COM188, baudrate=115200)                                                   
[1456529873.52][SERI][INF] reset device using 'default' plugin...                                                 
[1456529873.77][SERI][INF] wait for it...                                                                         
[1456529874.77][CONN][INF] sending preamble '91070f76-7bf9-4769-9b9f-b515b0a4e32e'...                             
[1456529874.77][SERI][TXD] mbedmbedmbedmbedmbedmbedmbedmbedmbedmbed                                               
[1456529874.77][SERI][TXD] {{__sync;91070f76-7bf9-4769-9b9f-b515b0a4e32e}}                                        
[1456529874.78][CONN][RXD] {{__sync;91070f76-7bf9-4769-9b9f-b515b0a4e32e}}                                        
[1456529874.78][CONN][INF] found SYNC in stream: {{__sync;91070f76-7bf9-4769-9b9f-b515b0a4e32e}}, queued...       
[1456529874.78][CONN][RXD] {{__timeout;120}}                                                                      
[1456529874.78][CONN][INF] found KV pair in stream: {{__timeout;120}}, queued...                                  
[1456529874.78][CONN][RXD] {{__host_test_name;default}}                                                           
[1456529874.78][CONN][INF] found KV pair in stream: {{__host_test_name;default}}, queued...                       
[1456529874.79][HTST][INF] sync KV found, uuid=91070f76-7bf9-4769-9b9f-b515b0a4e32e, timestamp=1456529874.781000  
[1456529874.79][HTST][INF] setting timeout to: 120 sec                                                            
[1456529874.79][HTST][INF] host test setup() call...                                                              
[1456529874.79][HTST][INF] CALLBACKs updated                                                                      
[1456529874.79][HTST][INF] host test detected: default                                                            
[1456529874.79][CONN][RXD] :48::FAIL: Expected TRUE Was FALSE                                                     
[1456529874.79][CONN][RXD] >>> failure with reason 'Assertion Failed' during 'Unknown Location'                   
[1456529874.79][CONN][INF] found KV pair in stream: {{__testcase_finish;;0;1}}, queued...                         
[1456529874.79][CONN][RXD] {{__testcase_finish;;0;1}}                                                             
[1456529874.80][CONN][RXD] >>> '': 0 passed, 1 failed with reason 'Assertion Failed'                              
[1456529874.80][CONN][RXD] >>> Test cases: 0 passed, 1 failed with reason 'Assertion Failed'                      
[1456529874.81][CONN][RXD] >>> TESTS FAILED!                                                                      
[1456529874.81][CONN][RXD] {{__testcase_summary;0;1}}                                                             
[1456529874.81][CONN][INF] found KV pair in stream: {{__testcase_summary;0;1}}, queued...                         
[1456529874.81][CONN][RXD] {{end;failure}}                                                                        
[1456529874.81][CONN][INF] found KV pair in stream: {{end;failure}}, queued...                                    
[1456529874.81][CONN][RXD] {{__exit;0}}                                                                           
[1456529874.81][CONN][INF] found KV pair in stream: {{__exit;0}}, queued...                                       
[1456529874.82][HTST][INF] __exit(0)                                                                              
[1456529874.82][HTST][INF] test suite run finished after 0.03 sec...                                              
[1456529874.82][HTST][INF] CONN exited with code: None                                                            
[1456529874.82][HTST][INF] 1 events in queue                                                                      
[1456529874.82][HTST][INF] __notify_complete(False)                                                               
[1456529874.82][HTST][INF] stopped consuming events                                                               
[1456529874.82][HTST][INF] host test result() call skipped, received: False                                       
[1456529874.82][HTST][INF] calling blocking teardown()                                                            
[1456529874.82][HTST][INF] teardown() finished                                                                    
[1456529874.82][HTST][INF] {{result;failure}}                                                                     
```

**Comment by ciarmcom** [comment]
ARM Internal Ref: IOTSFW-2141

**Comment by niklarm** [comment]
Hm, yes, I see how that can be an issue.

The design view was `utest -> unity`, so that you can exchange or supplement `unity` with additional assertion libraries.
However, the assertion library needs to communicate a failure back to `utest`, so somewhere something needed to be specialized.
In this case I made the decision to modify `unity` and make it work only with utest and put a warning into the Readme.

> Furthermore the failure macros have been modified to integrate with utest failure handlers

The issue are these [two modifications](https://github.com/ARMmbed/unity/blob/master/source/unity.c#L11-L12), which [calls `Harness::raise_failure()` in `utest`](https://github.com/ARMmbed/utest/blob/master/source/unity_handler.cpp#L24), which calls the default failure handler, which logs the failure.

> utest prints should not be turn on when we are not initializing harness!

Yes! I will add a check, so that `raise_failure()` only runs when the `Harness` has been initialized.

**Comment by niklarm** [comment]
I just released utest v1.9.2, can you check if that fixed the issue?
